src/preprocessing/passes/ackermann.cpp

```diff
@@ -187,9 +188,124 @@ void collectFunctionsAndLemmas(FunctionToArgsMap& fun_to_args,
 
 /* -------------------------------------------------------------------------- */
 
+bool needsReplace(TNode term)
+{
+  if (term.getType().isSort() && !(term.getKind() == kind::BOUND_VARIABLE))
+    return true;
+  return false;
+}
+
+/* Given the lowest capacity requirements for each uninterpreted sorts, assign
+ * a sufficient bit-vector size. Get the converting map */
+void collectUSortsToBV(USortToBVSizeMap& usortCardinality,
+                       vector<TNode>& vec,
+                       SubstitutionMap& sortsToSkolem)
+{
+  NodeManager* nm = NodeManager::currentNM();
+
+  for (TNode term : vec)
+  {
+    if (needsReplace(term))
+    {
+      TypeNode type = term.getType();
+      size_t size = usortCardinality[type].second;
+      if (size == 0)
+      {
+        size = log2(usortCardinality[type].first) + 1;
+        usortCardinality[type].second = size;
+      }
+      Node skolem = nm->mkSkolem(
+          "BVSKOLEM$$",
+          nm->mkBitVectorType(size),
+          "a variable created by the ackermannization "
+          "preprocessing pass, representing a term with uninterpreted sort.");
+      sortsToSkolem.addSubstitution(term, skolem);
+    }
+  }
+}
+
+/* This function returns the list of terms in formula represented by assertions.
+ * We use a BFS to get all terms without duplications. */
+std::vector<TNode> getListOfTerms(AssertionPipeline* assertions)
```

Comment:
Maybe you can use `getSymbols` from `node_algorithm.h` instead of this function?
https://github.com/CVC4/CVC4/blob/cbd86eb4ed8bafc17f28244b746a376a019462f1/src/expr/node_algorithm.h#L99

Reply:
Thanks for pointing. But getSymbols is get symbols from a term, while here getList is getting symbols for a formula. Seems they are different?

Reply:
Maybe I can use getVariables() for each assertion instead?

Reply:
https://github.com/CVC4/CVC4/blob/cbd86eb4ed8bafc17f28244b746a376a019462f1/src/expr/node_algorithm.cpp#L342

I think that `getSymbols` will work for every node. 
But I guess you can decide which one to use. Please make sure to look at their code (not only signature and comments) and also to make sure that you are not getting bound variables (there are not supposed to be any in your formulas).